source/frontend/src/components/common/mapFSM/useLoactionFeatureLoader.tsx

```diff
@@ -0,0 +1,101 @@
+import { PIMS_Property_Location_View } from '@/models/layers/pimsPropertyLocationView';
+
+export interface LocationFeatureDataset {
+  location: LatLngLiteral;
+  pimsFeature: Feature<Geometry, PIMS_Property_Location_View> | null;
+  parcelFeature: Feature<Geometry, PMBC_FullyAttributed_Feature_Properties> | null;
+  regionFeature: Feature<Geometry, MOT_RegionalBoundary_Feature_Properties> | null;
+  districtFeature: Feature<Geometry, MOT_DistrictBoundary_Feature_Properties> | null;
+  municipalityFeature: Feature<Geometry, WHSE_Municipalities_Feature_Properties> | null;
+}
+
+const useLoactionFeatureLoader = () => {
+  const fullyAttributedService = useFullyAttributedParcelMapLayer();
+  const adminBoundaryLayerService = useAdminBoundaryMapLayer();
+  const adminLegalBoundaryLayerService = useLegalAdminBoundariesMapLayer();
+
+  const {
+    loadProperties: { execute: loadProperties },
+  } = useMapProperties();
+
+  const fullyAttributedServiceFindOne = fullyAttributedService.findOne;
+  const adminBoundaryLayerServiceFindRegion = adminBoundaryLayerService.findRegion;
+  const adminBoundaryLayerServiceFindDistrict = adminBoundaryLayerService.findDistrict;
+  const adminLegalBoundaryLayerServiceFindOneMunicipality =
+    adminLegalBoundaryLayerService.findOneMunicipality;
+
+  const showLocationDetails = useCallback(
+    async (latLng: LatLngLiteral, isSelecting: boolean): Promise<LocationFeatureDataset> => {
+      const result: LocationFeatureDataset = {
+        location: latLng,
+        pimsFeature: null,
+        parcelFeature: null,
+        regionFeature: null,
+        districtFeature: null,
+        municipalityFeature: null,
+      };
+      try {
+        // call these APIs in parallel - notice there is no "await"
+        const fullyAttributedTask = fullyAttributedServiceFindOne(latLng);
+        const regionTask = adminBoundaryLayerServiceFindRegion(latLng, 'GEOMETRY');
+        const districtTask = adminBoundaryLayerServiceFindDistrict(latLng, 'GEOMETRY');
+
+        const parcelFeature = await fullyAttributedTask;
+        const regionFeature = await regionTask;
+        const districtFeature = await districtTask;
+        let pimsLocationProperties:
+          | FeatureCollection<Geometry, PIMS_Property_Location_View>
+          | undefined = undefined;
+
+        // Load PimsProperties
+        if (parcelFeature !== undefined) {
+          pimsLocationProperties = await loadProperties({
+            PID: parcelFeature.properties?.PID || '',
+            PIN: parcelFeature.properties?.PIN?.toString() || '',
+          });
+        }
+
+        // If not selecting and there is no other information availiable
+        if (!isSelecting) {
+          const municipalityFeature = await adminLegalBoundaryLayerServiceFindOneMunicipality(
+            latLng,
+          );
+
+          result.pimsFeature = pimsLocationProperties?.features[0] ?? null;
```

Comment:
can pimsLocationsProperties?.features never be an empty array?

Reply:
good catch. I;ve wrapped it in a check to make sure it does not cause issues